### ISSUE TITLE
False positive `closure_end_indentation` introduced in version 0.60.0

### ISSUE THREAD

**Issue by F1248** [realm/SwiftLint#6216]
### New Issue Checklist

- [X] I've Updated SwiftLint to the latest version.
- [X] I've searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues).

### Bug Description

As simplified as possible:

```swift
_ = { {
	false
} } // warning: Closure End Indentation Violation: expected 0, got 2 (closure_end_indentation)
```

Command:

```bash
$ swiftlint lint path/to/file.swift --config path/to/config.yml
```

### Environment

* SwiftLint version: 0.60.0
* Configuration file:

```yml
only_rules:
    - closure_end_indentation
```

**Comment by SimplyDanny** [comment]
I'd rather call this a fixed false negative. As the description of the rule states

> Closure end should have the same indentation as the line that started it.

the expectation would be

```swift
_ = { {
	false
}
}
```

because for both closures the starting line is `_ = { {`.

This is how the rule works in 0.60.0. I understand that isn't what you want, but I also fail to find a good condition as to why the rule wouldn't trigger here.

Note that

```swift
_ = { 
  { false }
}
```

and

```swift
_ = { 
  {
    false
  }
}
```

would not trigger the rule.

**Comment by F1248** [comment]
First of all: Thank you for not simply closing this as "not planned", but for allowing for discussion.

In my opinion, in the example, all four brackets have the same indentation of zero:

```
_ = { {
	false
} }
```

Although the second `}` is preceded by two characters `} ` (in the warning, SwiftLint reports the indentation as `got 2`), I would not consider it indented since the line it is on does not begin any with spaces or tabs.

I often use this when an argument is the result of a `switch` statement, which requires it to be wrapped in a closure.

```
enum AppleSecurityChip {

	case none
	case t1
	case t2
	case mSeries

	var name: String {
		String(localized: { switch self {
			case .none: "None"
			case .t1: "Apple T1"
			case .t2: "Apple T2"
			case .mSeries: "M-series"
		} }())
	}
}
```

**Comment by kapitoshka438** [comment]
@F1248 the rule says "Closure end should have the same indentation as **the line that started it.**" not the opening brace indentation.

@SimplyDanny there is a real false positive introduced in 0.60.0:
```
let var1 = var2
    .prop?.method {
    } // warning: Closure End Indentation Violation: expected 0, got 4 (closure_end_indentation)
```
Everything is fine when prop is not optional:
```
let var1 = var2
    .prop.method {
    } // no warning
```

**Comment by SimplyDanny** [comment]
> [@F1248](https://github.com/F1248) the rule says "Closure end should have the same indentation as **the line that started it.**" not the opening brace indentation.

That's true. My assumption is that the old implementation found the first `}` and since it is exactly indented like the starting line, it treated the code as valid. The new implementation correctly matches the second `}` which doesn't fit.

I really cannot come up with a good condition what would allow to ignore this case optionally.

As a side note, directly evaluated inline closures could be expensive. So you could also take this rule violation as a hint to refactor your code. 😉

@kapitoshka438: Your example is indeed a false positive.
